api/envoy/extensions/load_balancing_policies/least_request/v3/least_request.proto

```diff
@@ -22,10 +22,17 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // This configuration allows the built-in LEAST_REQUEST LB policy to be configured via the LB policy
 // extension point. See the :ref:`load balancing architecture overview
 // <arch_overview_load_balancing_types>` for more information.
-// [#next-free-field: 6]
 message LeastRequest {
+  // Available methods for selecting the host set from which to return the host with the
+  // fewest active requests.
+  enum SelectionMethod {
```

Comment:
Can you please add some guidance/recommendation on when full scan should be used?

Reply:
@barroca : Can you comment on your use case for full scan mode? I know it was different than mine. I'd like to mention both if applicable.

Reply:
I didn't have any use case for it :) I pickup the work since it looked simple enough for a learning opportunity to contribute to the project.

The use case I was trying to work on is when the number of hosts is very small and there is a probability that the selection of random hosts will lead to choosing the same one, or not choosing the one with least requests and leaving the load unbalanced. 

But this is mostly true when we are dealing with small number of requests, because when you consider high throughput the probability of repeating random selection decreases and we balance out requests between hosts (according the paper used for the original algorithm).

The second implementation, instead of doing a full scan from the same index from the host list was to randomise the start index so we reduce even more the chance of choosing the same index when the number of requests are equal. 

I guess the main usecase of a full scan is to guarantee that the request is sent to the host with least requests (which can be beneficial for some use cases like evenly splitting work on a map reduce).

Reply:
Just added a line that explains `N_CHOICES` is best for most scenarios, and also explained the niche scenarios in which `FULL_SCAN` may be preferable. Let me know what you think, @tonya11en and @ramaraochavali. (I think this is the last outstanding review comment)

Reply:
thank you